src/capi.rs

```diff
@@ -0,0 +1,147 @@
+use crate::api::{DriverType, Introspectable, Registers};
+use crate::driver::dummy::Dummy;
+#[cfg(feature = "hyper-v")]
+use crate::driver::hyperv::HyperV;
+#[cfg(feature = "kvm")]
+use crate::driver::kvm::Kvm;
+#[cfg(feature = "virtualbox")]
+use crate::driver::virtualbox::VBox;
+#[cfg(feature = "xen")]
+use crate::driver::xen::Xen;
+use crate::init;
+use cty::{c_char, size_t, uint16_t, uint64_t, uint8_t};
+use std::ffi::{c_void, CStr};
+use std::slice;
+
+#[repr(C)]
+pub struct MicrovmiContext {
+    driver: *mut c_void,
+    driver_type: DriverType,
+}
+
+#[repr(C)]
+pub enum MicrovmiStatus {
+    MicrovmiSuccess,
+    MicrovmiFailure,
+}
+
+#[allow(clippy::missing_safety_doc)]
+#[no_mangle]
+pub unsafe extern "C" fn microvmi_init(
+    domain_name: *const c_char,
+    driver_type: DriverType,
+) -> *mut MicrovmiContext {
+    let safe_domain_name = CStr::from_ptr(domain_name).to_string_lossy().into_owned();
+    let driver = init(&safe_domain_name, Some(driver_type.clone()));
```

Comment:
We should be able to pass a NULL `driver_type` here, so the `init` will attempt each driver until there is a match.
That's the default behavior in [mem-dump.rs](https://github.com/Wenzel/libmicrovmi/blob/master/examples/mem-dump.rs#L27)

So the user doesn't have to edit the file to change the `driver_type`.